### ISSUE TITLE
WIP Vdevel 3.13.0 (log refactor) - [merged]

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1678]
In GitLab by @dbeltrankyl on Aug 18, 2020, 11:50

_Merges vdevel-3.13.0 -> master_

this is where I'll continue the log_rework, the next step would be merged into master @mcastril.

The difference between this branch and log_rework is that this is also on modules (when I solve an issue with -v)

Edit: I'll resolve the comments of !187 on this one.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 18, 2020, 12:31

added 1 commit

<ul><li>cee6069e - # Version keyword force an exception in parse arg due and os_exit(0) but the...</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4725&start_sha=497c1099edefd05f3ef8a2b4e3fd3e8545b4dc97)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 18, 2020, 13:12

added 1 commit

<ul><li>4f8c61b6 - # Version keyword force an exception in parse arg due and os_exit(0) but the...</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4728&start_sha=cee6069e2ec61313d4fbc4f054565602cc668b71)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 18, 2020, 13:23

added 1 commit

<ul><li>f05e6840 - # Version keyword force an exception in parse arg due and os_exit(0) but the...</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4729&start_sha=4f8c61b635b781f61096fa9866092a7dce8b2ef8)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 18, 2020, 13:34

added 1 commit

<ul><li>ca4c60e0 - # Version keyword force an exception in parse arg due and os_exit(0) but the...</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4730&start_sha=f05e684044694aa7301f6e93110741cdc903f234)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 20, 2020, 13:27

added 1 commit

<ul><li>0d910dc9 - # Version keyword force an exception in parse arg due and os_exit(0) but the...</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4741&start_sha=ca4c60e0307a2f4dcb391943b98b8b161abe43f8)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 21, 2020, 15:27

added 1 commit

<ul><li>6112d055 - # Version keyword force an exception in parse arg due and os_exit(0) but the...</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4762&start_sha=0d910dc9046356aea0a78f655b26b61ace427aed)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 21, 2020, 15:54

added 1 commit

<ul><li>c8b7e027 - # Version keyword force an exception in parse arg due and os_exit(0) but the...</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4763&start_sha=6112d055bd874c3accf61c52743e0b659e21ec1e)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 21, 2020, 16:01

added 1 commit

<ul><li>3d068ad9 - disable check filepath in refresh</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4764&start_sha=c8b7e027ebbda0a840798179009cc33505539a6f)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 21, 2020, 16:08

added 1 commit

<ul><li>b6889c12 - disable check filepath in refresh</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4765&start_sha=3d068ad90b73cf365f49c4075cc97eef44343b0b)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 21, 2020, 16:52

added 1 commit

<ul><li>bd2e3618 - disable check filepath in refresh</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4768&start_sha=b6889c1280a45567dc8b246fd6dc7d3722faf98e)

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 24, 2020, 10:19

Commented on [autosubmit/database/db_common.py line 107](https://github.com/BSC-ES/autosubmit/compare/401f9dde84b181477a9ff80e392fb21366819881..d7edb893710ce4e257f58e7da6cd99da9f8b5b2d#diff-63842269881591045338affae9a6d5f8L107)



In messages, better to always use Autosubmit , upcased, unless it's the name of a parameter or variable.

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 24, 2020, 10:27

Commented on [autosubmit/job/job_packages.py line 96](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..96e868b83e4f0875b7deb09b20a1f3fb2a4b226e#diff-a7889a84c29dba8b1cee6ea5524fd736R96)



This was not corrected.

**Comment by kinow** [comment]
In GitLab by @mcastril on Aug 24, 2020, 10:27

Commented on [autosubmit/job/job_packages.py line 99](https://github.com/BSC-ES/autosubmit/compare/401f9dde84b181477a9ff80e392fb21366819881..d7edb893710ce4e257f58e7da6cd99da9f8b5b2d#diff-a7889a84c29dba8b1cee6ea5524fd736L99)



"Template using CHECK=On_submission has some empty variables" would be more clear.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 07:13

added 1 commit

<ul><li>2cf9d186 - little fix</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4803&start_sha=bd2e3618f157d932c73e8459309ee173455f38e2)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 09:56

added 1 commit

<ul><li>96e868b8 - little fix</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4804&start_sha=2cf9d1868c0d95c40949383d4f8f4217d4484b2e)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 14:12

Commented on [autosubmit/job/job_packages.py line 96](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..96e868b83e4f0875b7deb09b20a1f3fb2a4b226e#diff-a7889a84c29dba8b1cee6ea5524fd736R96)



changed this line in [version 14 of the diff](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4806&start_sha=96e868b83e4f0875b7deb09b20a1f3fb2a4b226e#8d19b0fb38faa36bd1780af738c2cb0dbd53a767_96_96)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 14:12

added 1 commit

<ul><li>4e54aade - Added codes</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4806&start_sha=96e868b83e4f0875b7deb09b20a1f3fb2a4b226e)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 15:05

added 1 commit

<ul><li>46979c1f - readthedocs code addition (critical ones)</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4807&start_sha=4e54aadedf49ada86bf403375ca63735549d3316)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 15:57

added 1 commit

<ul><li>3ddf26e8 - readthedocs code addition (critical ones)</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4808&start_sha=46979c1f6a334979250c121e4c3d616e93c75299)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 16:07

added 1 commit

<ul><li>913e0a53 - readthedocs code addition (critical ones)</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4809&start_sha=3ddf26e8864185f28ddc6ae5f7a3180e4722b12a)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 16:11

added 1 commit

<ul><li>a6d6b4af - readthedocs code addition (critical ones)</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4810&start_sha=913e0a531b51efcb14f17247ad88fb6393197ee0)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 31, 2020, 17:14

added 1 commit

<ul><li>2f932206 - readthedocs table</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4811&start_sha=a6d6b4af369cb991fc96e78eecf5b1b6a9416d0d)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 1, 2020, 10:51

added 1 commit

<ul><li>a37759f3 - readthedocs table error codes</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4813&start_sha=2f932206848b68dbf5675ced624dc9f08495934d)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 1, 2020, 13:18

added 1 commit

<ul><li>5d5857b5 - Error trace missing fix</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4817&start_sha=a37759f30465fb8ab5b64eb9a65acf08c490417e)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 2, 2020, 14:44

added 1 commit

<ul><li>dd3247be - Missing authentificatio nexception, and other tweaks to non mn4 platform.</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4823&start_sha=5d5857b5651690ae0788d535c5d11730da706ad5)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 12:01

added 1 commit

<ul><li>494e7f57 - Added log error on ecplatform</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4827&start_sha=dd3247bed40e4c50f2c0034e256cc91b0ebbaa4e)

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:47

Commented on [autosubmit/database/db_common.py line 320](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..bd2e3618f157d932c73e8459309ee173455f38e2#diff-63842269881591045338affae9a6d5f8R320)



This one should start with upcase too.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:49

Commented on [autosubmit/config/config_common.py line 359](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R359)



"contain", as files is a plural

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:50

Commented on [autosubmit/config/config_common.py line 361](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R361)



Should be "In", instead of "On"

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:50

Commented on [autosubmit/config/config_common.py line 361](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R361)



Also in "Issues on" in the next lines.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:51

Commented on [autosubmit/config/config_common.py line 371](https://github.com/BSC-ES/autosubmit/compare/401f9dde84b181477a9ff80e392fb21366819881..d7edb893710ce4e257f58e7da6cd99da9f8b5b2d#diff-6a739692a2475eda78e40c53adf27f41L371)



The following lines are little difficult to follow (the motivation of the try/excepts. Some comments could help.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:53

Commented on [autosubmit/config/config_common.py line 512](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R512)



It would be "File {0} doesn't exist"

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:54

Commented on [autosubmit/config/config_common.py line 514](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R514)



Here too.

The messages thrown in the logs must be checked. You know, only once written, but thousands of times shown and read by everyone.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:57

Commented on [autosubmit/database/db_common.py line 205](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-63842269881591045338affae9a6d5f8R205)



It is usually said: "Could not establish a connection to database". Wherever possible, use active voice, it's more clear.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:58

Commented on [autosubmit/git/autosubmit_git.py line 106](https://github.com/BSC-ES/autosubmit/compare/401f9dde84b181477a9ff80e392fb21366819881..d7edb893710ce4e257f58e7da6cd99da9f8b5b2d#diff-d7b47b8637947d763001053645b70febL106)



Here Git was moved to small letters (git), but then in line 127, "Git" appears again. The same convention should be respected everywhere.

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 3, 2020, 12:59

Commented on [autosubmit/job/job_list.py line 943](https://github.com/BSC-ES/autosubmit/compare/401f9dde84b181477a9ff80e392fb21366819881..d7edb893710ce4e257f58e7da6cd99da9f8b5b2d#diff-bad488e977334d250f2df049877c45f3R943)



It's a good idea to put TIPs and etc. 

And sometimes a link to the FAQ could be helpful: https://autosubmit.readthedocs.io/en/latest/faq.html

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:49

Commented on [autosubmit/config/config_common.py line 514](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R514)



>The messages thrown in the logs must be checked. You know, only once written, but thousands of times shown and read by everyone.

I don't get this, all of these messages are shown only once with the function

They are store in self.wrong_config or self.warn_config ( depending on the gravity of the issue) 

And in
config_common.py
line 405 result = self.show_messages()

Will be printed by file, section, parameter and message

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:54

Commented on [autosubmit/job/job_list.py line 943](https://github.com/BSC-ES/autosubmit/compare/401f9dde84b181477a9ff80e392fb21366819881..d7edb893710ce4e257f58e7da6cd99da9f8b5b2d#diff-bad488e977334d250f2df049877c45f3R943)



I'm adding the faq link at the end of execution if autosubmit has detected a critical issue. Although now it is called error codes and solutions. Faq did only contain solutions to the errors so I think the faq should be another different thing but I don't know

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:58

Commented on [autosubmit/config/config_common.py line 359](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R359)



changed this line in [version 24 of the diff](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4829&start_sha=494e7f57b62c83e76236bd4da536bc75efbba624#4c5196a3ff59501fdbf099c26091d37248bce779_359_359)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:58

Commented on [autosubmit/config/config_common.py line 361](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R361)



changed this line in [version 24 of the diff](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4829&start_sha=494e7f57b62c83e76236bd4da536bc75efbba624#4c5196a3ff59501fdbf099c26091d37248bce779_361_361)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:58

Commented on [autosubmit/config/config_common.py line 512](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R512)



changed this line in [version 24 of the diff](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4829&start_sha=494e7f57b62c83e76236bd4da536bc75efbba624#4c5196a3ff59501fdbf099c26091d37248bce779_512_515)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:58

Commented on [autosubmit/config/config_common.py line 514](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R514)



changed this line in [version 24 of the diff](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4829&start_sha=494e7f57b62c83e76236bd4da536bc75efbba624#4c5196a3ff59501fdbf099c26091d37248bce779_514_517)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:58

Commented on [autosubmit/database/db_common.py line 205](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-63842269881591045338affae9a6d5f8R205)



changed this line in [version 24 of the diff](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4829&start_sha=494e7f57b62c83e76236bd4da536bc75efbba624#a0717de6e7c89084c02c0488de5a7833d15908d4_205_205)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:58

added 1 commit

<ul><li>d7edb893 - Fixed changes mentioned by miguel, fixed logs mechanism retrieval ( and now is working much better)</li></ul>

[Compare with previous version](/gitlab/es/autosubmit/-/merge_requests/190/diffs?diff_id=4829&start_sha=494e7f57b62c83e76236bd4da536bc75efbba624)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 3, 2020, 13:58

I think that now all works, I'll proceed to resolve the merge conflict and it will be finished

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 4, 2020, 13:02

merged

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Sep 4, 2020, 13:17

mentioned in commit e83f97d875d4baf48464ee430ae5e7b21b0ace48

**Comment by kinow** [comment]
In GitLab by @mcastril on Sep 4, 2020, 17:40

Commented on [autosubmit/config/config_common.py line 514](https://github.com/BSC-ES/autosubmit/compare/20ac837b66a9526ab355027d0465852fa170b54a..494e7f57b62c83e76236bd4da536bc75efbba624#diff-6a739692a2475eda78e40c53adf27f41R514)



Sorry, this was not about any technical issue. My point is that you code the message once, but then user's see the message thousands of times (counting all potential users, all the times they run the code, hoping there's not a lot of failures).

So the point was to ensure everything is grammatically correct.
